modules/mcaptcha.nix

```
@@ -0,0 +1,279 @@
+{
+  config,
+  lib,
+  options,
+  pkgs,
+  ...
+}:
+with builtins;
+with lib; let
```

Nice explicit way of inheriting from `lib`:
```suggestion
let
  inherit (lib) mkIf /* ... */;
```